### PR TITLE
IEEE 215 *new* checked out order card

### PR DESCRIPTION
**Pull request by natapokie**
## Overview

- Resolves 215 Create Checked Out/Picked Up Order Card
- Had some issues with running test so made a new branch from #415
- New order card called `OrderPending`
  - clickable order card that evokes an alert event with the order ID displayed
  - adding picked up and view buttons
  - once picked up is clicked, button is diabled along with the checkbox
- Note: changes from another member are attached to this branch

## Unit Tests Created

- front end test to make sure details are displayed on card

## Steps to QA

- use the mockPendingOrders from mock data and call the OrderPending component in the Orders page instead of allOrders
- clicking on the card will evoke the alert
- clicking on picked up disables itself and the checkbox

image

### CODE REVIEW

File: hackathon_site/dashboard/frontend/src/components/orders/OrderCard/OrderPending.test.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        const date = new Date(time);
+        const month = date.toLocaleString("default", { month: "short" });
+        const day = date.getDate();
+        const hoursAndMinutes = date.getHours() + ":" + date.getMinutes();
+
+        const orderDetails = [
+            { title: "Team", value: teamCode },
+            { title: "Order Qty", value: orderQuantity },
+            { title: "Time ordered", value: `${month} ${day}, ${hoursAndMinutes}` },
+            // { title: "Received IDs", value: receivedIDs },
```
Comment (Mustaballer) [inline hackathon_site/dashboard/frontend/src/components/orders/OrderCard/OrderPending.test.tsx]: Should remove if not being used

File: hackathon_site/dashboard/frontend/src/components/orders/OrderCard/OrderPending.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    };
+
+    return (
+        <Card onClick={alertEvent}>
+            <Formik
+                initialValues={{
+                    ReceivedIDsCheck: false,
+                }}
+                onSubmit={(values: FormikValues) => {
+                    console.log(values);
```
Comment (Mustaballer) [inline hackathon_site/dashboard/frontend/src/components/orders/OrderCard/OrderPending.tsx]: Please remove logging in the final version :)

File: hackathon_site/dashboard/frontend/src/components/orders/OrderCard/OrderPending.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+                            justifyContent: "space-between",
+                            alignItems: "center",
+                            padding: "4px 12px",
+                        }}
+                    >
+                        <Button
+                            type="submit"
+                            onClick={(event) => {
+                                event.stopPropagation();
+                                // console.log("click picked up");
```
Comment (Mustaballer) [inline hackathon_site/dashboard/frontend/src/components/orders/OrderCard/OrderPending.tsx]: remove comment

File: hackathon_site/dashboard/frontend/src/components/orders/OrderCard/OrderPending.tsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+                            className={stylesButton.filterBtnsApply}
+                            disabled={pickedUp}
+                            disableElevation
+                            data-testid="apply-button"
+                            style={pickedUp ? { pointerEvents: "none" } : {}}
+                        >
+                            Picked Up
+                        </Button>
+                        <Button
+                            // type="submit"
```
Comment (Mustaballer) [inline hackathon_site/dashboard/frontend/src/components/orders/OrderCard/OrderPending.tsx]: remove comment

File: hackathon_site/dashboard/frontend/src/components/orders/OrderCard/OrderPending.tsx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+                            color="primary"
+                            data-testid="clear-button"
+                            onClick={(event) => {
+                                event.stopPropagation();
+                                console.log("click view");
```
Comment (Mustaballer) [inline hackathon_site/dashboard/frontend/src/components/orders/OrderCard/OrderPending.tsx]: pls remove log

File: hackathon_site/dashboard/frontend/src/components/orders/OrderCard/OrderPending.tsx [hunk 6]
```diff
@@ -0,0 +1,140 @@
+import React, { useEffect, useState } from "react";
+import styles from "./OrderCardPending.module.scss";
+import { Typography, Container, Card } from "@material-ui/core";
+import { Formik, FormikValues } from "formik";
+import Checkbox from "@material-ui/core/Checkbox";
+import Button from "@material-ui/core/Button";
+import stylesButton from "components/sharedStyles/Filter.module.scss";
+
+interface OrderPendingProps {
+    team: string;
```
Comment (Mustaballer) [inline hackathon_site/dashboard/frontend/src/components/orders/OrderCard/OrderPending.tsx]: This should be team code and should still be a string

**Review by Mustaballer — CHANGES_REQUESTED** [review]
Natalie, great job! I'm amazed by the quality of your code especially since its your first PR in the codebase. I left some minor comments below, next steps would be to use real data. Before we delve into that, I do want to ask you some questions regarding the designs.

1. Do you think its immediately obvious the difference between `Picked Up` and `Ready for Pickup` orders? If not you can look into modifying the design to be more obvious.
2. Right now we have custom order models for `Picked Up` and `Ready for Pickup`, are you interested in desigining new Order cards for other statuses.(We could also make updates to the current Order Cards) 

Overall we're pretty flexible with the design, so do let me know what you think of the OrderCards and how we can make them both simple and easy to undestand.

**Comment by natapokie** [comment]
> Natalie, great job! I'm amazed by the quality of your code especially since its your first PR in the codebase. I left some minor comments below, next steps would be to use real data. Before we delve into that, I do want to ask you some questions regarding the designs.
> 
> 1. Do you think its immediately obvious the difference between `Picked Up` and `Ready for Pickup` orders? If not you can look into modifying the design to be more obvious.
> 2. Right now we have custom order models for `Picked Up` and `Ready for Pickup`, are you interested in desigining new Order cards for other statuses.(We could also make updates to the current Order Cards)
> 
> Overall we're pretty flexible with the design, so do let me know what you think of the OrderCards and how we can make them both simple and easy to undestand.

Thank you! But I only made adjustments to the code that was written previously haha, to answer your questions...

1. To be completely honest, as someone who had no prior experience with the HSS I didn't immediately figure out which design was the `Picked Up` and `Ready for Pickup` orders. One thing that I would like to propose is actually using a table instead of an order card? To me the orders system feels really similar to Shopify where there are different statuses for items for example...
    - "Unfulfilled" means the order is not ready
    - "Ready for Pickup" 
    - "Fulfiled" when the order has been picked up

  Kinda like this...!
  image

I think the table would allow us to easily incorporate the other "states" like pending and even cancelled orders and this way we wouldn't have to make too many components that are really similar...! And another idea, we can probably have a state for 
"Preparing", I remember when I was helping out last year, we ended up doing an order that someone else was already preparing.
Let me know what you think about that... and I can mock up a separate design on Figma ;)

And also for 2. yes, I can make more order cards as well if we decide to go that route!

**Comment by Mustaballer** [comment]
Hey Natalie, I really like the idea of using a table for handling orders! I think it'd be pretty straightforward in figuring out what orders need to be done, and help us manage them more effectively. I think it'd be great if you could mock up a design on Figma :)
Regarding the preparing status, I also think that's a good idea, but we should be considering how to make it work in real time since we have to refresh the HSS when we want fetch the latest data. We could do some more brainstorming later(I'm thinking something to do with Socket.IO)

Final note, you can close this PR, and work on designing a mockup of what the Order Tables could look like :)

**Pull request closed without merging**